cloud-sql/mysql/client-side-encryption/src/main/java/com/example/cloudsql/EncryptAndInsertData.java

```diff
@@ -0,0 +1,122 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.cloudsql;
+
+// [START cloud_sql_mysql_encrypt_insert]
+import com.google.crypto.tink.aead.AeadConfig;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.KmsClients;
+import com.google.crypto.tink.aead.AeadKeyTemplates;
+import com.google.crypto.tink.integration.gcpkms.GcpKmsClient;
+import com.google.crypto.tink.proto.KeyTemplate;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.security.GeneralSecurityException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import javax.sql.DataSource;
+
+class EncryptAndInsertData {
+
+  public static void main(String[] args) throws GeneralSecurityException, SQLException {
+    // TODO(developer): Replace these variables before running the sample.
+    String dbUser = "your-username"; // e.g. "root", "mysql"
+    String dbPass = "your-password";
+    String dbName = "your-database-name";
+    String cloudSqlConnectionName = "project:region:instance";
+    String kmsUri = "gcp-kms://" + "your-kms-uri";
+    String team = "TABS";
+    String tableName = "votes";
+    String email = "hello@example.com";
+
+    // Initialize database connection pool and create table
+    DataSource pool = createConnectionPool(dbUser, dbPass, dbName, cloudSqlConnectionName);
+    createTable(pool, tableName);
+
+    // Initialize envelope AEAD
+    Aead envAead = getEnvelopeAead(kmsUri);
+
+    encryptAndInsertData(pool, envAead, tableName, team, email);
+  }
+
+  public static void encryptAndInsertData(DataSource pool, Aead envAead, String tableName,
+      String team, String email)
+      throws GeneralSecurityException, SQLException {
+
+    try (Connection conn = pool.getConnection()) {
+      String stmt = String.format(
+          "INSERT INTO %s (team, time_cast, voter_email) VALUES (?, ?, ?);", tableName);
+      try (PreparedStatement voteStmt = conn.prepareStatement(stmt);) {
+        voteStmt.setString(1, team);
+        voteStmt.setTimestamp(2, new Timestamp(new Date().getTime()));
+
+        // Use the envelope AEAD primitive to encrypt the email, using the team name as
+        // associated data
+        byte[] encrypted_email = envAead.encrypt(email.getBytes(), team.getBytes());
+        voteStmt.setBytes(3, encrypted_email);
+
+        // Finally, execute the statement. If it fails, an error will be thrown.
+        voteStmt.execute();
+        System.out.println(String.format("Successfully inserted row into table %s", tableName));
+      }
+    }
+  }
+
+  public static Aead getEnvelopeAead(String kmsUri) throws GeneralSecurityException {
+    AeadConfig.register();
+
+    // Generate a new envelope key template, then generate key material.
```

Comment:
We actually have a plan to deprecate this envelope encryption API.

Here's the new way (which is identical to Python):

KmsClient client = new GcpKmsClient().withDefaultCredentials();

Aead remote = client.getAead(kmsUri);

return new KmsEnvelopeAead(AeadKeyTemplates.AES128_GCM, remote);